lib/ecto/repo/queryable.ex

```diff
@@ -235,6 +235,9 @@ defmodule Ecto.Repo.Queryable do
         {map, row}
     end
   end
+  defp process(row, {:map, {:source, :from}, args}, nil, _prefix, _adapter) do
+    {Map.new(args), row}
```

Comment:
Should the recursive [call here](https://github.com/AndrewDryga/ecto/blob/3eb6b5d4e7f4b4a72e29b35bd88cf1ef08620674/lib/ecto/repo/queryable.ex#L242) handle this case?

Reply:
It did not work, because from is nil. And that call is basically calling [this code](https://github.com/AndrewDryga/ecto/blob/3eb6b5d4e7f4b4a72e29b35bd88cf1ef08620674/lib/ecto/repo/queryable.ex#L256-L258):
```
  defp process(row, {:source, :from}, from, _prefix, _adapter) do
    {from, row}
  end
```

But here args knowledge is lost, so within this clause we can't really do anything.

Reply:
Ok, I think we need to generalize this a bit. I will merge and play with it. :) Thank you! :heart:

Reply:
You're welcome. And in case you feel like guiding me won't take more time than fixing it by yourself, I can take care of it :). Just need to know a place to start.

Reply:
I would love to guide but I still have no idea what needs to be done or if something needs to be done. I want to understand why we are passing `nil` in the first place. :)